pkg/client/unversioned/deployment.go

```diff
@@ -94,12 +94,13 @@ func (c *deployments) Update(deployment *extensions.Deployment) (result *extensi
 }
 
 // Watch returns a watch.Interface that watches the requested deployments.
-func (c *deployments) Watch(label labels.Selector, field fields.Selector, resourceVersion string) (watch.Interface, error) {
+func (c *deployments) Watch(label labels.Selector, field fields.Selector, opts api.ListOptions) (watch.Interface, error) {
 	return c.client.Get().
 		Prefix("watch").
 		Namespace(c.ns).
 		Resource("deployments").
-		Param("resourceVersion", resourceVersion).
+		Param("resourceVersion", opts.ResourceVersion).
```

Comment:
I think removing it from special params is what we want.

On Tue, Oct 20, 2015 at 10:42 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/client/unversioned/deployment.go
> <https://github.com/kubernetes/kubernetes/pull/15705#discussion_r42504303>
> :
>
> >  	return c.client.Get().
> >  		Prefix("watch").
> >  		Namespace(c.ns).
> >  		Resource("deployments").
> > -		Param("resourceVersion", resourceVersion).
> > +		Param("resourceVersion", opts.ResourceVersion).
>
> Got it - thanks.
> One more question. I've added "timeoutSeconds" to special params here:
>
> https://github.com/kubernetes/kubernetes/pull/15705/files#diff-fa79e9ef069aba6022a27bd97ab2c0beR48
> which means that VersionedParams will not set it according to the current
> code:
>
> https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/request.go#L424
>
> If we want to use it I need to remove "timeoutSeconds" from special params
> - is that what we want? ("timeout" is the special param).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15705/files#r42504303>.
>